ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -0,0 +1,1306 @@
+
+                        # Remove vlans if upstream switch does not have them #}
+                        switch_vlans = self._vlans
+                        uplink_switch_vlans = uplink_switch_facts._vlans
+                        uplink_vlans = list(set(switch_vlans).intersection(uplink_switch_vlans))
+
+                        if self.inband_management_vlan is not None:
+                            uplink_vlans.append(int(self.inband_management_vlan))
+
+                        uplink['vlans'] = self._list_compress(uplink_vlans)
+
+                        if self.short_esi is not None:
+                            uplink['peer_short_esi'] = self.short_esi
+
+                        if self.link_tracking_groups is not None:
+                            uplink['link_tracking_groups'] = []
+                            for lt_group in self.link_tracking_groups:
+                                uplink['link_tracking_groups'].append({"name": lt_group["name"], "direction": "upstream"})
+
+                        uplinks.append(uplink)
+        return uplinks
+
+    @cached_property
+    def uplink_peers(self):
+        '''
+        List of all uplink peers
+
+        This are used to generate the "avd_topology_peers" fact covering downlinks for all devices.
+        '''
+        if self.uplinks is not None:
+            return [uplink['peer'] for uplink in self.uplinks]
+
+    @cached_property
+    def _mlag_peer_id(self):
+        if self.mlag is True:
+            return get(self._hostvars, f"avd_switch_facts.{self.mlag_peer}.switch.id", required=True)
+
+    @cached_property
+    def vtep_ip(self):
+        '''
+        Run template lookup to render ipv4 address for vtep_ip
+
+        Since some templates might contain certain legacy variables (switch_*),
+        those are mapped from the switch.* model
+        '''
+        if self.vtep is True:
+            template_vars = {"ansible_search_path": self._ansible_search_path}
+            # Copying __dict__ will expose all switch facts cached up until this function is run.
+            # TODO: We should probably find and document a list of supported context variables instead.
+            template_vars['switch'] = {k: self.__dict__.get(k) for k in self.keys()}
+            template_vars['switch_id'] = self.id
+            template_vars['switch_vtep_loopback_ipv4_pool'] = self.vtep_loopback_ipv4_pool
+            template_vars['loopback_ipv4_offset'] = self.loopback_ipv4_offset
+
+            if self.mlag is True:
+                if self.mlag_role == 'primary':
+                    template_vars['mlag_primary_id'] = self.id
+                    template_vars['mlag_secondary_id'] = self._mlag_peer_id
+                elif self.mlag_role == 'secondary':
+                    template_vars['mlag_secondary_id'] = self.id
+                    template_vars['mlag_primary_id'] = self._mlag_peer_id
```

Comment:
```suggestion
            if self.mlag is True:
                template_vars['mlag_primary_id'] = self.id if self.mlag_role == "primary" else self._mlag_peer_id
                template_vars['mlag_secondary_id'] = self._mlag_peer_id if self.mlag_role == "primary" else self.id
```

Only a suggestion - not everyone loves ternary operators
though in any case the case where role is neither `primary` not `secondary` is not handled (but would be handled by schema)

Reply:
We normally prefer the more elaborate style.